src/base/bittorrent/session.cpp

```diff
@@ -2997,6 +3082,16 @@ void Session::handleTorrentCategoryChanged(TorrentHandle *const torrent, const Q
     emit torrentCategoryChanged(torrent, oldCategory);
 }
 
+void Session::handleTorrentTagAdded(const QString &tag, int new_tag_count)
+{
+    emit torrentTagAdded(tag, new_tag_count);
```

Comment:
The interface of the object should be meaningful regardless of who uses it and how. In your case, the Session "says": my torrent added (removed) tag, but does not indicate on this torrent. Look at other similar signals and do your same. Besides, you must omit "new tag count" parameter, because the observer can obtain it from the torrent itself (`torrent->tags().count()`).